packages/nextcloud/test/cookbook_test.dart

```diff
@@ -0,0 +1,193 @@
+        client = await TestNextcloudClient.create(container);
+      });
+
+      tearDown(() async {
+        await container.destroy();
+      });
+
+      group('CookbookVersionCheck', () {
+        test('Is supported', () async {
+          final response = await client.cookbook.getVersionCheck();
+
+          expect(response.isSupported, isTrue);
+        });
+      });
+
+      group('categories', () {
+        test('listCategories', () async {
+          final response = await client.cookbook.categories.listCategories();
+
+          expect(response.body, hasLength(7));
+        });
+
+        test('recipesInCategory', () async {
+          final response = await client.cookbook.categories.recipesInCategory(category: 'Soup');
+
+          expect(response.body, hasLength(2));
+        });
+
+        test('renameCategory', () async {
+          final request = cookbook.RenameCategoryRequestApplicationJsonBuilder()..name = 'Delicious Soup';
+
+          final response = await client.cookbook.categories.renameCategory(
+            category: 'Soup',
+            $body: request.build(),
+          );
+
+          expect(response.body, request.name);
+        });
+      });
+
+      group('misc', () {
+        test('getConfig', () async {
+          final response = await client.cookbook.misc.getConfig();
+          final visibleInfoBlocks = cookbook.VisibleInfoBlocksBuilder()
+            ..preparationTime = true
+            ..cookingTime = true
+            ..totalTime = true
+            ..nutritionInformation = true
+            ..tools = true;
+          final expected = cookbook.ConfigBuilder()
+            ..folder = '/Recipes'
+            ..updateInterval = 5
+            ..printImage = true
+            ..visibleInfoBlocks = visibleInfoBlocks;
+
+          expect(response.body, equalsBuilt(expected.build()));
+        });
+
+        test('reindex', () async {
+          final response = await client.cookbook.misc.reindex();
+
+          expect(response.body, 'Search index rebuilt successfully');
+        });
+
+        test('setConfig', () async {
+          final config = cookbook.ConfigBuilder()..folder = '/';
+          final response = await client.cookbook.misc.setConfig($body: config.build());
+
+          expect(response.body, 'OK');
+        });
+
+        test('version', () async {
+          final response = await client.cookbook.misc.version();
+
+          final cookbookVersion = response.body.cookbookVersion!;
+          expect(cookbookVersion[0], JsonObject(preset.version.major));
+          expect(cookbookVersion[1], JsonObject(preset.version.minor));
+        });
+      });
+
+      group('recipes', () {
+        test('callImport', () async {
+          final url = cookbook.UrlBuilder()
+            ..url = 'https://www.chefkoch.de/rezepte/126221054029392/Germknoedel-mit-Powidl.html';
+          final response = await client.cookbook.recipes.$import($body: url.build());
```

Comment:
I don't like that it depends on an external service that could go down or change. Can you download the whole page and put it into the static folder?

Reply:
I understand the issue but in that case I'd rather create a custom site.

I don't think we can easily add their work into a gpl licenced codebase.

Reply:
Right, that's a good point. Then we simply have to hope it never breaks.